h2o-core/src/main/java/water/server/ServletUtils.java

```diff
@@ -173,6 +175,21 @@ public static String getDecodedUri(HttpServletRequest request) {
       throw new RuntimeException(e);
     }
   }
+  
+  public static String[] parseUriParams(String uri, HttpServletResponse response, Pattern p, int numParams) throws IOException {
+    Matcher m = p.matcher(uri);
+    if (!m.matches()) {
```

Comment:
I'd suggesting verifying the number of matches stored inside the matcher equals the expected number of params. Or derive the number of expected params from the number of matches if that does not need to be checked.